src/main/java/net/fabricmc/loom/LoomGradleExtension.java

```diff
@@ -114,7 +114,7 @@ default List<Path> getMinecraftJars(MappingsNamespace mappingsNamespace) {
 			yield getSrgMinecraftProvider().getMinecraftJarPaths();
 		}
 		case MOJANG -> {
-			ModPlatform.assertPlatform(this, ModPlatform.NEOFORGE, () -> "Mojang-mapped jars are only available on NeoForge.");
+			assert this.isForgeLike() && this.getForgeProvider().usesMojangAtRuntime() : "Mojang-mapped jars are only available on Forge 50+ or NeoForge.";
```

Comment:
Using `assert` here is a behaviour change (since this check needs assertions at runtime which are never enabled for Gradle probably?)